src/software/thunderscope/gl/layers/gl_validation_layer.py

```diff
@@ -111,79 +112,75 @@ def __update_validation_graphics(self, validations: List[ValidationProto]):
         :param validations: The list of validation protos
 
         """
-        polygon_graphics_index = 0
-        circle_graphics_index = 0
-
-        for validation in validations:
+        polygons = [
+            (polygon, validation.status)
+            for validation in validations
+            for polygon in validation.geometry.polygons
+        ]
+        segments = [
+            (segment, validation.status)
+            for validation in validations
+            for segment in validation.geometry.segments
+        ]
+        circles = [
+            (circle, validation.status)
+            for validation in validations
+            for circle in validation.geometry.circles
+        ]
+
+        # Ensure we have the same number of graphics as validations
+        self.polygon_graphics.resize(
+            len(polygons), lambda: GLLinePlotItem(width=3.0),
+        )
+        self.segment_graphics.resize(
+            len(segments), lambda: GLLinePlotItem(width=3.0),
+        )
+        self.circle_graphics.resize(
+            len(circles), lambda: GLCircle(),
+        )
```

Comment:
On my computer, the validation shapes can be difficult to see given that they are drawn right on top of the already existing white field lines/goal. I think we can improve it by filling in the shapes (i.e. Polygon, Circle) with the validation color. Though this is not a high priority task and doesn't have to be in this PR, we can create a ticket for it and track it there. For now, I say lets get this PR merged in.

Reply:
Added ability to fill GLRect or GLCircle with color + added GLPolygon which produces a filled graphic of any polygon